java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -322,26 +301,18 @@ public TypeParameterTreeImpl newTypeParameter(JavaTree extendsToken, BoundListTr
     return new TypeParameterTreeImpl((InternalSyntaxToken) extendsToken, bounds);
   }
 
-  public BoundListTreeImpl newBounds(TypeTree classType, Optional<List<AstNode>> rests) {
+  public BoundListTreeImpl newBounds(TypeTree classType, Optional<List<Tuple<InternalSyntaxToken, Tree>>> rests) {
     ImmutableList.Builder<Tree> classTypes = ImmutableList.builder();
-    List<AstNode> children = Lists.newArrayList();
 
     classTypes.add(classType);
-    children.add((AstNode) classType);
 
     if (rests.isPresent()) {
-      for (AstNode rest : rests.get()) {
-        for (AstNode child : rest.getChildren()) {
-          if (!child.is(JavaPunctuator.AND)) {
-            classTypes.add((Tree) child);
-          }
-
-          children.add(child);
-        }
+      for (Tuple<InternalSyntaxToken, Tree> rest : rests.get()) {
+        //FIXME SONARJAVA-547 comma should be handled (rest.first())
+          classTypes.add(rest.second());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 9. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)